### ISSUE TITLE
Account issues

### ISSUE THREAD

**Issue by AurelienDupont** [bgo-bioimagerie/platformmanager#99]
Dear all,

I am an engineer from MRic-TEM, Biosit, Rennes. I have one platformemanager account which is "admin" for this facility. The username is "Aurélien Dupont". 

I still can access to every menus but I cannot register new sessions on our devices... I think it started in July.

Would it be possible to help me? I don't really know how to fix it.

Thanks in advance,

Best regards.

**Comment by osallou** [comment]
what do you mean by "register new sessions" ?

**Comment by AurelienDupont** [comment]
I just cannot book new sessions on the microscope.
I open the calendar, click on "+" to book it. 
I have access to the "Modification Réservation" page. 
I can fill this page but when I click on "save" the same page reload and that's all. And no booking is done on the calendar.
I only have this problem since July. Before I could book all our devices from the facility.

**Comment by osallou** [comment]
looking at information on your account in database, your are not set as admin for this space (MRic-TEM)

**Comment by osallou** [comment]
you are inactive in our "backup" (old data), but in production database, you appear as admin, so everything should be ok. A reservation in our "dev" environement, which is a copy, is ok for me.

**Comment by osallou** [comment]
could you provide a screenshot of filled reservation page?

**Comment by AurelienDupont** [comment]
Yes apparently I am the only one... My colleagues, Agnès Burel who is also admin can book some sessions and also one user of our facility did it several times. This is why I really don't understand.
I don't know how to share one picture on this website. I hope it will work.

![capture d ecran 2018-09-05 a 16 48 10](https://user-images.githubusercontent.com/42338181/45101643-4343e300-b12c-11e8-9163-5498d1809e5a.png)

**Comment by osallou** [comment]
I see that reservation is in the past, 29/08.
Past reservations are allowed?

**Comment by AurelienDupont** [comment]
In the past I could for sure. I did it several times when it was "last time" reservetions.

I've just tried to book for next week and it's exactly the same.

![capture d ecran 2018-09-06 a 09 41 42](https://user-images.githubusercontent.com/42338181/45142681-636db380-b1b9-11e8-9efa-ca688b41a93f.png)

**Comment by osallou** [comment]
I gonna do some testing under your identity for testing

**Comment by osallou** [comment]
I did some code modifications, issue seems link to the resource using "packages" (don't really know what it is).
There were some errors in browser (hidden) and could fix them.
Could you try again a reservation?

**Comment by AurelienDupont** [comment]
I guess it's not a problem if no-one has already booked the microscope.

**Comment by osallou** [comment]
this is a problem with your user config, you have a reference to a "client" with a non existent id. Don't know the reason. I am looking how to solve this

**Comment by osallou** [comment]
I guess you need to activate "client" module in admin section, and create a client (which may be yourself and add youself as an authorized user)

I do not see however how other collegues can book on your platform as there is no client at all.

**Comment by AurelienDupont** [comment]
Well, thank you very much for your help and your time. This is really weird that it changed at a certain point.

**Comment by osallou** [comment]
could you test with a client defined?

**Comment by AurelienDupont** [comment]
The thing is, I don't really want to touch something in those menus. I am not sure about how it's working and I don't want to mess everything up. 
Usually, when I had one issue, I went to see Sylvain to know what to do.
For sure we have some users even without the modul "Client". For exemple "Ophélie Nicole" is register into 3 facilities. One is MRic TEM. From my understanding, this module is mainly used for facturation. And we do not use PlatformManager for this. We are only using it for the booking system.
If I check the "Utilisateurs" menu, I have acces to MRic TEM for sure. The same as others who can book our devices.
But perhaps I am totally wrong.

**Comment by osallou** [comment]
getting assigned to a platform is one thing, but according to the code, for the reservation, you need to be linked to a client (which can be yourself, or a -- fake client).
I do not know how the software works or is expected to work, we only host the software.

What I can see when logged with your user is reservation fails because it does not find any client linked to your account.

There may be other options/config/... to disable this for your platform, but we have no knowledge on the software itself.

**Comment by AurelienDupont** [comment]
Ok, thank you very much for all those explanations and this help ! I will try later this thing about the "client" when I will be back on the facility.

**Comment by osallou** [comment]
we faced same issue on an other plaform, and setting client solved the issue
